### PR TITLE
Exposes Injector

### LINKED ISSUE #12: Expose injector for use in integration tests
Is there a way to set up an integration test environment with some combination of dropwizard's `DropwizardAppRule` that would expose the injector used to start the service, so that you could grab an instance of an object (say a Dao)?

### PR DESCRIPTION
**Pull request by jaredculp**
Fixes #12

### CODE REVIEW

**Comment by jaredculp** [comment]
@jhaber let me know what you think of this! I noticed that the injector was exposed like this in the [previous version](https://github.com/HubSpot/dropwizard-guice/blob/master/src/main/java/com/hubspot/dropwizard/guice/GuiceBundle.java#L145) of this library. If there was a specific reason that that same decision was not made here please let me know. Hopefully the included `TestEnv` and `InjectorRule` classes show what I'm trying to accomplish.

**Comment by jhaber** [comment]
Sorry for the delay, I'll try to get this reviewed today

File: pom.xml [hunk 1]
```diff
@@ -98,6 +98,16 @@
       <groupId>com.fasterxml.jackson.core</groupId>
       <artifactId>jackson-databind</artifactId>
     </dependency>
+    <dependency>
+      <groupId>io.dropwizard</groupId>
+      <artifactId>dropwizard-testing</artifactId>
```
Comment (jhaber) [inline pom.xml]: Compile scope?

File: pom.xml [hunk 2]
```diff
@@ -98,6 +98,16 @@
...
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.12</version>
+      <scope>compile</scope>
```
Comment (jhaber) [inline pom.xml]: Compile scope?

File: src/main/java/com/hubspot/dropwizard/guicier/GuiceBundle.java [hunk 3]
```diff
@@ -115,6 +116,10 @@ protected Injector getInjector() {
     });
   }
 
+  public Injector getInjector() {
+    return this.injector;
```
Comment (jhaber) [inline src/main/java/com/hubspot/dropwizard/guicier/GuiceBundle.java]: What about 
```java
return checkNotNull(injector, "injector has not been initialized yet");
```

File: src/main/java/com/hubspot/dropwizard/guicier/GuicierApplication.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * Generally speaking, you should store instance variables for the GuiceBundle and Injector.
+ * You'll want to initialize the GuiceBundle field as part of your initialize() method and
+ * you'll want to initialize the Injector field as part of your run() method. This is due to
+ * the way dropwizard initializes the Environment and Configuration objects.
+ *
+ * The injector can then be exposed via the getInjector() method you are required to implement.
+ * @param <T> the type of Configuration for the application
+ */
+public abstract class GuicierApplication<T extends Configuration> extends Application<T> {
```
Comment (jhaber) [inline src/main/java/com/hubspot/dropwizard/guicier/GuicierApplication.java]: Would it be possible to drop this class and just require that the class implements `Supplier<Injector>` or something to that effect?

File: pom.xml (hunk 1 above)
Comment (jaredculp) [inline pom.xml]: Yes because we're using these in the `src/main` code rather than the `src/test` code.

**Comment by jaredculp** [comment]
@jhaber thanks for the review! I think your intuition about the test code living in a separate library is the right call. I've backed out those changes and now this just contains the changes to expose the injector in `GuiceBundle`.

**Comment by jhaber** [comment]
Awesome, thanks. If I remember correctly there may be some subtlety with actually wanting to expose a child injector but we can start with this and see if there are any issues

**Comment by jaredculp** [comment]
🙌🏻 any chance you can cut a release that we can depend on internally?

**Comment by jhaber** [comment]
Sure thing, I'll get that done today

**Comment by jhaber** [comment]
Just released version 1.0.0.5, should show up in central soon

**Pull request merged**